synchro/__init__.py

```diff
@@ -483,7 +495,7 @@ def __getattr__(self, name):
 
     def __getitem__(self, name):
         # Access to collections with dotted names, like db.test['mike']
-        fullname = self.name + "." + name
+        fullname = self.name + "." + str(name)
```

Comment:
I don't think we want to call str(name) since name should already be a str.

Reply:
Same comment as the core class

Reply:
We should skip those tests. Motor should not accept non-strings for getitem even if pymongo does.